enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+| `.spec.buildInputs.imageBuilder`                  | Required                  | Which image builder backend to use. Right now, the only option is `customPodBuilder`. Future backends, such as OpenShift Pipelines (aka Tekton) may be enabled by changing this option.                                                                                       |
+| `.spec.buildInputs.renderedImagePushSecret`       | Required                  | Name of the secret required to push the final OS image to an image registry.                                                                                                                                                                                                  |
+| `.spec.buildInputs.renderedImagePushspec`         | Required                  | The tagged image pushspec (`registry.hostname.com/org/repo:tag`) where the final built image will be used to. Note: Any user-supplied tags (such as `:latest`) will not be honored.                                                                                           |
+| `.spec.buildInputs.containerfiles`                | Required                  | The Containerfile(s) to build for each OS image. Multiple Containerfiles will be allowed to account for minor discrepancies for multiarch builds. Initially, this was a field on the MachineOSConfig object, but this could be a reference (or references) to a ConfigMap(s). |
+| `.spec.buildOutputs.currentImagePullSecret`       | Required                  | Name of image pull secret that will be used by each node to pull the built image.                                                                                                                                                                                             |
+| `.spec.buildOutputs.postBuildStrategy`            | Optional                  | What should happen after the build is complete. Initially, options should be something like, `ImagePushOnly`, `ImagePushAndRollout`, which are self-explanatory. Alternatively, this could be an annotation as described below.                                               |
+
+MachineOSConfigs can also be augmented with the following annotations:
+
+| **Annotation Name:**                             | **Description:**                                                                                                                           |
+| ------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------ |
+| `machineconfiguration.openshift.io/rebuildImage` | When present, this will clear any failed image builds (and any supporting transitory objects created) and retry the build.                 |
```

Comment:
Make this a field in the spec, `redbuildReason`, and then in the status, `lastRebuildReason`, if they differ, a new build is triggered and the rebuilder reason is copied to status on success